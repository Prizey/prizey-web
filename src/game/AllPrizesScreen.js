import React from 'react'
import { List as CroodsList } from 'croods'
import { Link } from '@reach/router'
import { Typography } from '@material-ui/core'
import { withStyles } from '@material-ui/core/styles'
import List from '@material-ui/core/List'
import Avatar from '@material-ui/core/Avatar'
import ListItem from '@material-ui/core/ListItem'

import ListItemText from '@material-ui/core/ListItemText'
import ListItemAvatar from '@material-ui/core/ListItemAvatar'
import ListSubheader from '@material-ui/core/ListSubheader'
import CircularProgress from '@material-ui/core/CircularProgress'

import Layout from 'design/Layout/Layout'
import UserBalance from 'design/UserBalance'
import ProfileLink from 'design/ProfileLink/ProfileLink'

const difficulties = [
  { label: 'EASY', name: 'easy' },
  { label: 'MEDIUM', name: 'medium' },
  { label: 'HARD', name: 'hard' },
]

const styles = theme => ({
  listItem: {
    background: 'rgba(0, 0, 0, 0.087)',
    borderBottom: '1px solid rgba(255, 255, 255, 0.087)',
  },
  login: {
    align: 'left',
    marginRight: '20px',
    [theme.breakpoints.down('xs')]: {
      marginRight: 0,
    },
  },
  root: {
    borderRadius: 6,
    height: '65vh',
    overflow: 'auto',
  },
  sectionList: {
    margin: 0,
    padding: 0,
  },
  subheader: {
    fontWeight: 'bold',
    marginTop: theme.spacing.xs,
  },
})

const AllPrizesScreen = withStyles(styles)(
  ({ classes, location, currentUser }) => (
    <Layout
      leftIcon={<UserBalance />}
      rightIcon={
        currentUser ? (
          <ProfileLink />
        ) : (
          <Link to={'/sign-in?next=/game'} style={{ textDecoration: 'none' }}>
            <Typography className={classes.login}>Login</Typography>
          </Link>
        )
      }
      location={location}
      currentUser={currentUser}
    >
      <List subheader={<li />} className={classes.root}>
        {difficulties.map(difficulty => (
          <li key={`section-${difficulty.name}`}>
            <ul className={classes.sectionList}>
              <ListSubheader className={classes.subheader} disableSticky>
                {difficulty.label}
              </ListSubheader>
              <CroodsList
                parentId={difficulty.name}
                name="products"
                path={`/products/${difficulty.name}`}
                renderLoading={() => (
                  <CircularProgress
                    color="inherit"
                    size={30}
                    className={classes.loading}
                  />
                )}
                render={products =>
                  products.map(item => (
                    <ListItem
                      key={`item-${difficulty.name}-${item.id}`}
                      className={classes.listItem}
                    >
                      <ListItemAvatar>
                        <Avatar src={item.image} alt={item.title} />
                      </ListItemAvatar>
                      <ListItemText
                        primary={item.title}
                        secondary={`MSRP: $${parseFloat(item.price).toFixed(
                          2,
                        )}`}
                      />
                    </ListItem>
                  ))
                }
              />
            </ul>
          </li>
        ))}
      </List>
    </Layout>
  ),
)

export default AllPrizesScreen
