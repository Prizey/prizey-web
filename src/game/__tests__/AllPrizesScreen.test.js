import React from 'react'
import renderer from 'react-test-renderer'
import AllPrizesScreen from '../AllPrizesScreen'

const mockList = [
  { id: 1, image: '/mocks/trump-mask.png', title: 'trump' },
  { id: 2, image: '/mocks/sweatshirt.png', title: 'sweatshirt' },
  { id: 3, image: '/mocks/shoe.png', title: 'shoe' },
]

jest.mock('../../design/Layout/RegisterPageView')
jest.mock('design/UserBalance', () => () => null)

jest.mock('react-redux', () => ({
  connect: () => Component => props => (
    <Component {...props} insertCoin={jest.fn} />
  ),
}))

jest.mock('croods', () => ({
  Info: ({ children, ...props }) => (
    <div {...props}>
      Info -{' '}
      {props.render({
        easyTicketAmount: 1,
        hardicketAmount: 2,
        mediumTicketAmount: 3,
      })}
    </div>
  ),
  List: props => (
    <div {...props}>
      {props.children} - Render: {props.render(mockList)}
      RenderLoading: {props.renderLoading()}
    </div>
  ),
  New: props => (
    <div {...props}>
      New - <div>render - {props.render(props)}</div>
    </div>
  ),
  Provider: ({ children, ...props }) => (
    <div {...props}>Provider - {children}</div>
  ),
  createReducer: () => (state = {}) => state,
}))

it('renders correctly', () => {
  const params = {
    currentUser: {
      tickets: 10,
    },
  }
  const tree = renderer.create(<AllPrizesScreen {...params} />).toJSON()
  expect(tree).toMatchSnapshot()
})
