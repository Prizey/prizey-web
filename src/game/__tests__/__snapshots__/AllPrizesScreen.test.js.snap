// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Component-root-13"
>
  <div
    className="Component-header-11"
  >
    <a
      aria-label="My profile"
      className="MuiButtonBase-root-21 MuiIconButton-root-15 Component-root-14"
      href="/profile"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="MuiIconButton-label-20"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-24 MuiSvgIcon-fontSizeLarge-32"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <g>
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.07 18.28c.43-.9 3.05-1.78 4.93-1.78s4.51.88 4.93 1.78C15.57 19.36 13.86 20 12 20s-3.57-.64-4.93-1.72zm11.29-1.45c-1.43-1.74-4.9-2.33-6.36-2.33s-4.93.59-6.36 2.33C4.62 15.49 4 13.82 4 12c0-4.41 3.59-8 8-8s8 3.59 8 8c0 1.82-.62 3.49-1.64 4.83z"
            />
            <path
              d="M12 6c-1.94 0-3.5 1.56-3.5 3.5S10.06 13 12 13s3.5-1.56 3.5-3.5S13.94 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
            />
          </g>
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-113"
      />
    </a>
  </div>
  <div
    className="Component-container-8"
  >
    <a
      aria-current="page"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <img
        alt="Prizey"
        className="Component-logoImage-12"
        src="/logo.png"
      />
    </a>
    <div
      className="Component-caption-6"
    />
    <div
      className="Component-children-7"
    >
      <ul
        className="MuiList-root-33 MuiList-padding-34 MuiList-subheader-36 Component-root-3"
      >
        <li />
        <li>
          <ul
            className="Component-sectionList-4"
          >
            <li
              className="MuiListSubheader-root-37 MuiListSubheader-gutters-40 Component-subheader-5"
            >
              EASY
            </li>
            <div
              name="products"
              parentId="easy"
              path="/products/easy"
              render={[Function]}
              renderLoading={[Function]}
            >
               - Render: 
              <li
                className="MuiListItem-root-43 MuiListItem-default-46 MuiListItem-dense-47 MuiListItem-gutters-51 Component-listItem-1"
                disabled={false}
              >
                <div
                  className="MuiAvatar-root-58"
                >
                  <img
                    alt="trump"
                    className="MuiAvatar-img-60"
                    src="/mocks/trump-mask.png"
                  />
                </div>
                <div
                  className="MuiListItemText-root-61"
                >
                  <span
                    className="MuiTypography-root-67 MuiTypography-subheading-74 MuiListItemText-primary-64"
                  >
                    trump
                  </span>
                  <p
                    className="MuiTypography-root-67 MuiTypography-body1-76 MuiTypography-colorTextSecondary-100 MuiListItemText-secondary-65"
                  >
                    MSRP: $NaN
                  </p>
                </div>
              </li>
              <li
                className="MuiListItem-root-43 MuiListItem-default-46 MuiListItem-dense-47 MuiListItem-gutters-51 Component-listItem-1"
                disabled={false}
              >
                <div
                  className="MuiAvatar-root-58"
                >
                  <img
                    alt="sweatshirt"
                    className="MuiAvatar-img-60"
                    src="/mocks/sweatshirt.png"
                  />
                </div>
                <div
                  className="MuiListItemText-root-61"
                >
                  <span
                    className="MuiTypography-root-67 MuiTypography-subheading-74 MuiListItemText-primary-64"
                  >
                    sweatshirt
                  </span>
                  <p
                    className="MuiTypography-root-67 MuiTypography-body1-76 MuiTypography-colorTextSecondary-100 MuiListItemText-secondary-65"
                  >
                    MSRP: $NaN
                  </p>
                </div>
              </li>
              <li
                className="MuiListItem-root-43 MuiListItem-default-46 MuiListItem-dense-47 MuiListItem-gutters-51 Component-listItem-1"
                disabled={false}
              >
                <div
                  className="MuiAvatar-root-58"
                >
                  <img
                    alt="shoe"
                    className="MuiAvatar-img-60"
                    src="/mocks/shoe.png"
                  />
                </div>
                <div
                  className="MuiListItemText-root-61"
                >
                  <span
                    className="MuiTypography-root-67 MuiTypography-subheading-74 MuiListItemText-primary-64"
                  >
                    shoe
                  </span>
                  <p
                    className="MuiTypography-root-67 MuiTypography-body1-76 MuiTypography-colorTextSecondary-100 MuiListItemText-secondary-65"
                  >
                    MSRP: $NaN
                  </p>
                </div>
              </li>
              RenderLoading: 
              <div
                className="MuiCircularProgress-root-103 MuiCircularProgress-colorPrimary-106 MuiCircularProgress-indeterminate-105"
                role="progressbar"
                style={
                  Object {
                    "height": 30,
                    "width": 30,
                  }
                }
              >
                <svg
                  className="MuiCircularProgress-svg-108"
                  viewBox="22 22 44 44"
                >
                  <circle
                    className="MuiCircularProgress-circle-109 MuiCircularProgress-circleIndeterminate-111"
                    cx={44}
                    cy={44}
                    fill="none"
                    r={20.2}
                    strokeWidth={3.6}
                    style={Object {}}
                  />
                </svg>
              </div>
            </div>
          </ul>
        </li>
        <li>
          <ul
            className="Component-sectionList-4"
          >
            <li
              className="MuiListSubheader-root-37 MuiListSubheader-gutters-40 Component-subheader-5"
            >
              MEDIUM
            </li>
            <div
              name="products"
              parentId="medium"
              path="/products/medium"
              render={[Function]}
              renderLoading={[Function]}
            >
               - Render: 
              <li
                className="MuiListItem-root-43 MuiListItem-default-46 MuiListItem-dense-47 MuiListItem-gutters-51 Component-listItem-1"
                disabled={false}
              >
                <div
                  className="MuiAvatar-root-58"
                >
                  <img
                    alt="trump"
                    className="MuiAvatar-img-60"
                    src="/mocks/trump-mask.png"
                  />
                </div>
                <div
                  className="MuiListItemText-root-61"
                >
                  <span
                    className="MuiTypography-root-67 MuiTypography-subheading-74 MuiListItemText-primary-64"
                  >
                    trump
                  </span>
                  <p
                    className="MuiTypography-root-67 MuiTypography-body1-76 MuiTypography-colorTextSecondary-100 MuiListItemText-secondary-65"
                  >
                    MSRP: $NaN
                  </p>
                </div>
              </li>
              <li
                className="MuiListItem-root-43 MuiListItem-default-46 MuiListItem-dense-47 MuiListItem-gutters-51 Component-listItem-1"
                disabled={false}
              >
                <div
                  className="MuiAvatar-root-58"
                >
                  <img
                    alt="sweatshirt"
                    className="MuiAvatar-img-60"
                    src="/mocks/sweatshirt.png"
                  />
                </div>
                <div
                  className="MuiListItemText-root-61"
                >
                  <span
                    className="MuiTypography-root-67 MuiTypography-subheading-74 MuiListItemText-primary-64"
                  >
                    sweatshirt
                  </span>
                  <p
                    className="MuiTypography-root-67 MuiTypography-body1-76 MuiTypography-colorTextSecondary-100 MuiListItemText-secondary-65"
                  >
                    MSRP: $NaN
                  </p>
                </div>
              </li>
              <li
                className="MuiListItem-root-43 MuiListItem-default-46 MuiListItem-dense-47 MuiListItem-gutters-51 Component-listItem-1"
                disabled={false}
              >
                <div
                  className="MuiAvatar-root-58"
                >
                  <img
                    alt="shoe"
                    className="MuiAvatar-img-60"
                    src="/mocks/shoe.png"
                  />
                </div>
                <div
                  className="MuiListItemText-root-61"
                >
                  <span
                    className="MuiTypography-root-67 MuiTypography-subheading-74 MuiListItemText-primary-64"
                  >
                    shoe
                  </span>
                  <p
                    className="MuiTypography-root-67 MuiTypography-body1-76 MuiTypography-colorTextSecondary-100 MuiListItemText-secondary-65"
                  >
                    MSRP: $NaN
                  </p>
                </div>
              </li>
              RenderLoading: 
              <div
                className="MuiCircularProgress-root-103 MuiCircularProgress-colorPrimary-106 MuiCircularProgress-indeterminate-105"
                role="progressbar"
                style={
                  Object {
                    "height": 30,
                    "width": 30,
                  }
                }
              >
                <svg
                  className="MuiCircularProgress-svg-108"
                  viewBox="22 22 44 44"
                >
                  <circle
                    className="MuiCircularProgress-circle-109 MuiCircularProgress-circleIndeterminate-111"
                    cx={44}
                    cy={44}
                    fill="none"
                    r={20.2}
                    strokeWidth={3.6}
                    style={Object {}}
                  />
                </svg>
              </div>
            </div>
          </ul>
        </li>
        <li>
          <ul
            className="Component-sectionList-4"
          >
            <li
              className="MuiListSubheader-root-37 MuiListSubheader-gutters-40 Component-subheader-5"
            >
              HARD
            </li>
            <div
              name="products"
              parentId="hard"
              path="/products/hard"
              render={[Function]}
              renderLoading={[Function]}
            >
               - Render: 
              <li
                className="MuiListItem-root-43 MuiListItem-default-46 MuiListItem-dense-47 MuiListItem-gutters-51 Component-listItem-1"
                disabled={false}
              >
                <div
                  className="MuiAvatar-root-58"
                >
                  <img
                    alt="trump"
                    className="MuiAvatar-img-60"
                    src="/mocks/trump-mask.png"
                  />
                </div>
                <div
                  className="MuiListItemText-root-61"
                >
                  <span
                    className="MuiTypography-root-67 MuiTypography-subheading-74 MuiListItemText-primary-64"
                  >
                    trump
                  </span>
                  <p
                    className="MuiTypography-root-67 MuiTypography-body1-76 MuiTypography-colorTextSecondary-100 MuiListItemText-secondary-65"
                  >
                    MSRP: $NaN
                  </p>
                </div>
              </li>
              <li
                className="MuiListItem-root-43 MuiListItem-default-46 MuiListItem-dense-47 MuiListItem-gutters-51 Component-listItem-1"
                disabled={false}
              >
                <div
                  className="MuiAvatar-root-58"
                >
                  <img
                    alt="sweatshirt"
                    className="MuiAvatar-img-60"
                    src="/mocks/sweatshirt.png"
                  />
                </div>
                <div
                  className="MuiListItemText-root-61"
                >
                  <span
                    className="MuiTypography-root-67 MuiTypography-subheading-74 MuiListItemText-primary-64"
                  >
                    sweatshirt
                  </span>
                  <p
                    className="MuiTypography-root-67 MuiTypography-body1-76 MuiTypography-colorTextSecondary-100 MuiListItemText-secondary-65"
                  >
                    MSRP: $NaN
                  </p>
                </div>
              </li>
              <li
                className="MuiListItem-root-43 MuiListItem-default-46 MuiListItem-dense-47 MuiListItem-gutters-51 Component-listItem-1"
                disabled={false}
              >
                <div
                  className="MuiAvatar-root-58"
                >
                  <img
                    alt="shoe"
                    className="MuiAvatar-img-60"
                    src="/mocks/shoe.png"
                  />
                </div>
                <div
                  className="MuiListItemText-root-61"
                >
                  <span
                    className="MuiTypography-root-67 MuiTypography-subheading-74 MuiListItemText-primary-64"
                  >
                    shoe
                  </span>
                  <p
                    className="MuiTypography-root-67 MuiTypography-body1-76 MuiTypography-colorTextSecondary-100 MuiListItemText-secondary-65"
                  >
                    MSRP: $NaN
                  </p>
                </div>
              </li>
              RenderLoading: 
              <div
                className="MuiCircularProgress-root-103 MuiCircularProgress-colorPrimary-106 MuiCircularProgress-indeterminate-105"
                role="progressbar"
                style={
                  Object {
                    "height": 30,
                    "width": 30,
                  }
                }
              >
                <svg
                  className="MuiCircularProgress-svg-108"
                  viewBox="22 22 44 44"
                >
                  <circle
                    className="MuiCircularProgress-circle-109 MuiCircularProgress-circleIndeterminate-111"
                    cx={44}
                    cy={44}
                    fill="none"
                    r={20.2}
                    strokeWidth={3.6}
                    style={Object {}}
                  />
                </svg>
              </div>
            </div>
          </ul>
        </li>
      </ul>
    </div>
    <div
      className="Component-footer-9"
    >
      <a
        className="Component-footerLink-10"
        href="https://app.termly.io/document/terms-and-conditions/4de5f3bb-a333-47bb-9daf-f82f7d4e1cfb"
        rel="noopener noreferer"
        target="_blank"
      >
        <span
          className="MuiTypography-root-67 MuiTypography-caption-77 MuiTypography-colorTextSecondary-100 MuiTypography-alignCenter-90"
        >
          TERMS
        </span>
      </a>
      <a
        className="Component-footerLink-10"
        href="https://app.termly.io/document/privacy-policy/8192f107-a2dc-4548-bf83-97907c6d83c3"
        rel="noopener noreferer"
        target="_blank"
      >
        <span
          className="MuiTypography-root-67 MuiTypography-caption-77 MuiTypography-colorTextSecondary-100 MuiTypography-alignCenter-90"
        >
          PRIVACY
        </span>
      </a>
      <a
        className="Component-footerLink-10"
        href="/faq"
        onClick={[Function]}
        rel="noopener noreferer"
        target="_blank"
      >
        <span
          className="MuiTypography-root-67 MuiTypography-caption-77 MuiTypography-colorTextSecondary-100 MuiTypography-alignCenter-90"
        >
          FAIRNESS / FAQ
        </span>
      </a>
      <a
        className="Component-footerLink-10"
        href="/all-prizes"
        onClick={[Function]}
        rel="noopener noreferer"
        target="_blank"
      >
        <span
          className="MuiTypography-root-67 MuiTypography-caption-77 MuiTypography-colorTextSecondary-100 MuiTypography-alignCenter-90"
        >
          ALL PRIZES
        </span>
      </a>
    </div>
  </div>
</div>
`;
